### ISSUE TITLE
Receiving JSON data from java over to python

### ISSUE THREAD

**Issue by Maddy-robos** [kivy/pyjnius#452]
I have tried to receive JSONObject from java over to python. I tried converting it into string (normal non-pretty) and sending it over to jnius autoclass. It raises an exception stating `jnius.JavaException: JVM exception occurred: Missing value at 18 [character 0 line 2]`. The same output is proper in the Java file. For Example I am receiving a string output:

> {"AllDocuments":[{"Document":{"Negations":[],"AAs":[],"Utterances":[{"Phrases":[{"PhraseText":"A spinal tap","PhraseLength":"12","Candidates":[],"SyntaxUnits":[{"LexCat":"det","SyntaxType":"det","LexMatch":"a","InputMatch":"A","Tokens":["a"]},{"LexCat":"noun","SyntaxType":"head","LexMatch":"spinal tap","InputMatch":"spinal tap","Tokens":["spinal","tap"]}],"Mappings":[{"MappingScore":"-1000","MappingCandidates":[{"Status":"0","IsOverMatch":"no","MatchedWords":["spinal","tap"],"IsHead":"yes","CandidateMatched":"Spinal tap","CandidateCUI":"C0553794","Sources":["AOD","CCS","CCS_10","CHV","ICD9CM","MTH","NLMSubSyn","SNMI","SNOMEDCT_US"],"CandidatePreferred":"Diagnostic lumbar puncture","SemTypes":["diap"],"MatchMaps":[{"LexVariation":"0","TextMatchEnd":"2","ConcMatchStart":"1","TextMatchStart":"1","ConcMatchEnd":"2"}],"CandidateScore":"-1000","ConceptPIs":[{"StartPos":"2","Length":"10"}],"Negated":"0"}]}],"PhraseStartPos":"0"},{"PhraseText":"was","PhraseLength":"3","Candidates":[],"SyntaxUnits":[{"LexCat":"aux","SyntaxType":"aux","LexMatch":"was","InputMatch":"was","Tokens":["was"]}],"Mappings":[],"PhraseStartPos":"13"},{"PhraseText":"performed","PhraseLength":"9","Candidates":[],"SyntaxUnits":[{"LexCat":"verb","SyntaxType":"verb","LexMatch":"performed","InputMatch":"performed","Tokens":["performed"]}],"Mappings":[],"PhraseStartPos":"17"},{"PhraseText":"and","PhraseLength":"3","Candidates":[],"SyntaxUnits":[{"LexCat":"conj","SyntaxType":"conj","LexMatch":"and","InputMatch":"and","Tokens":["and"]}],"Mappings":[],"PhraseStartPos":"27"},{"PhraseText":"oligoclonal bands","PhraseLength":"17","Candidates":[],"SyntaxUnits":[{"LexCat":"noun","SyntaxType":"head","LexMatch":"oligoclonal bands","InputMatch":"oligoclonal bands","Tokens":["oligoclonal","bands"]}],"Mappings":[],"PhraseStartPos":"31"},{"PhraseText":"were","PhraseLength":"4","Candidates":[],"SyntaxUnits":[{"LexCat":"aux","SyntaxType":"aux","LexMatch":"were","InputMatch":"were","Tokens":["were"]}],"Mappings":[],"PhraseStartPos":"49"},{"PhraseText":"detected in the cerebrospinal fluid.","PhraseLength":"36","Candidates":[],"SyntaxUnits":[{"LexCat":"verb","SyntaxType":"verb","LexMatch":"detected","InputMatch":"detected","Tokens":["detected"]},{"LexCat":"prep","SyntaxType":"prep","LexMatch":"in","InputMatch":"in","Tokens":["in"]},{"LexCat":"det","SyntaxType":"det","LexMatch":"the","InputMatch":"the","Tokens":["the"]},{"LexCat":"noun","SyntaxType":"mod","LexMatch":"cerebrospinal fluid","InputMatch":"cerebrospinal fluid","Tokens":["cerebrospinal","fluid"]},{"SyntaxType":"punc","InputMatch":".","Tokens":[]}],"Mappings":[{"MappingScore":"-806","MappingCandidates":[{"Status":"0","IsOverMatch":"no","MatchedWords":["cerebro","spinal","fluid"],"IsHead":"yes","CandidateMatched":"Spinal Fluid, Cerebro","CandidateCUI":"C0007806","Sources":["AOD","CHV","CSP","CST","FMA","HL7V2.5","LCH","LCH_NW","LNC","MSH","MTH","NCI","NCI_CDISC","NCI_NCI-GLOSS","NLMSubSyn","SNM","SNMI","SNOMEDCT_US","UWDA"],"CandidatePreferred":"Cerebrospinal Fluid","SemTypes":["bdsu"],"MatchMaps":[{"LexVariation":"0","TextMatchEnd":"5","ConcMatchStart":"1","TextMatchStart":"4","ConcMatchEnd":"3"}],"CandidateScore":"-806","ConceptPIs":[{"StartPos":"70","Length":"19"}],"Negated":"0"}]},{"MappingScore":"-761","MappingCandidates":[{"Status":"0","IsOverMatch":"no","MatchedWords":["detected"],"IsHead":"yes","CandidateMatched":"Detected","CandidateCUI":"C0442726","Sources":["CHV","HL7V3.0","LNC","MTH","SNOMEDCT_US"],"CandidatePreferred":"Detected (finding)","SemTypes":["fndg"],"MatchMaps":[{"LexVariation":"0","TextMatchEnd":"1","ConcMatchStart":"1","TextMatchStart":"1","ConcMatchEnd":"1"}],"CandidateScore":"-760","ConceptPIs":[{"StartPos":"54","Length":"8"}],"Negated":"0"},{"Status":"0","IsOverMatch":"no","MatchedWords":["cerebro","spinal","fluid"],"IsHead":"yes","CandidateMatched":"Spinal Fluid, Cerebro","CandidateCUI":"C0007806","Sources":["AOD","CHV","CSP","CST","FMA","HL7V2.5","LCH","LCH_NW","LNC","MSH","MTH","NCI","NCI_CDISC","NCI_NCI-GLOSS","NLMSubSyn","SNM","SNMI","SNOMEDCT_US","UWDA"],"CandidatePreferred":"Cerebrospinal Fluid","SemTypes":["bdsu"],"MatchMaps":[{"LexVariation":"0","TextMatchEnd":"5","ConcMatchStart":"1","TextMatchStart":"4","ConcMatchEnd":"3"}],"CandidateScore":"-806","ConceptPIs":[{"StartPos":"70","Length":"19"}],"Negated":"0"}]},{"MappingScore":"-760","MappingCandidates":[{"Status":"0","IsOverMatch":"no","MatchedWords":["detected"],"IsHead":"yes","CandidateMatched":"Detected","CandidateCUI":"C0442726","Sources":["CHV","HL7V3.0","LNC","MTH","SNOMEDCT_US"],"CandidatePreferred":"Detected (finding)","SemTypes":["fndg"],"MatchMaps":[{"LexVariation":"0","TextMatchEnd":"1","ConcMatchStart":"1","TextMatchStart":"1","ConcMatchEnd":"1"}],"CandidateScore":"-760","ConceptPIs":[{"StartPos":"54","Length":"8"}],"Negated":"0"}]}],"PhraseStartPos":"54"}],"UttSection":"tx","UttNum":"1","UttStartPos":"0","UttText":"A spinal tap was performed and oligoclonal bands were detected in the cerebrospinal fluid.","UttLength":"90","PMID":"00000000"}],"CmdLine":{"Options":[{"OptName":"lexicon","OptValue":"db"},{"OptName":"mm_data_year","OptValue":"2018AA"},{"OptName":"ignore_word_order"},{"OptName":"all_derivational_variants"},{"OptName":"JSONn"},{"OptName":"negex"},{"OptName":"word_sense_disambiguation"},{"OptName":"strict_model"},{"OptName":"ignore_stop_phrases"},{"OptName":"exclude_sts","OptValue":["resa","tmco","qlco","ftcn","hlca","gngm","acty","podg","famg","orch","phsu","inpr","geoa","qnco","cnce","clna","orgf","aapp","dora","spco","vita","lbtr","lbpr","inbe","bhvr","ocac","clas","inch","medd","socb","idcn","anst","anim","ffas","ocdi","hcro","mnob","humn","elii","cgab","blor","topp","menp","aggp"]},{"OptName":"infile","OptValue":"user_input"},{"OptName":"outfile","OptValue":"user_output"}],"Command":"metamap16.BINARY.Linux --lexicon db -Z 2018AA -iD --JSONn --negex -y -A -K -k resa,tmco,qlco,ftcn,hlca,gngm,acty,podg,famg,orch,phsu,inpr,geoa,qnco,cnce,clna,orgf,aapp,dora,spco,vita,lbtr,lbpr,inbe,bhvr,ocac,clas,inch,medd,socb,idcn,anst,anim,ffas,ocdi,hcro,mnob,humn,elii,cgab,blor,topp,menp,aggp"}}}]}

 in the Java class. I converted it into a jar file with dependencies and have used the same in the python jnius library by setting a CLASSPATH environment variable. The library is receiving the class files but is not getting an output. I couldn't find any appropriate solution to this problem nor do I understand the problem. Any help is much appreciated.

**Comment by tito** [comment]
Could you provide a reproducible example? I don't understand exactly what you're doing. Thanks you

**Comment by Maddy-robos** [comment]
For Example, I tried sending the json string in the first comment from Java using a return string statement. Then I used a python class to print the string. I was able to print only `{"AllDocuments":[` Now I assumed the problem is a newline character in the json string. So I converted the whole json string into a JSONObject in java and converted it back to string which should make it a normal string with no newline characters. When I tried this, I kept getting an error `jnius.JavaException: JVM exception occurred: Missing value at 18 [character 0 line 2]`. Then I tried sending the JSONObject as a JSONObject and it gave the same error above.

How I achieved this is, I compiled a java file and converted it as an artifact jar file. I used this jar file in the python to call the class I created.

Now I was not able to send the string over in pretty-print json format, normal json format, or as a JSONObject (Java) over to python. Hope this clears it.
